### DISCUSSION TITLE
Tools for Skybox

### DISCUSSION THREAD

**Discussion by soltsmank** [o3de/o3de#6806] [Q&A] [answered]
Hello,

What tools is used to create the Skybox (*.streamingimage) file out of panorama HDR or EXR picture ?

Thanks,
(Sorry, if I missed existing guide on a topic. Please point if so.)

**Comment by HogJonny-AMZN** [comment, marked as answer]
Take a look here: C:\Depot\o3de-dev\Gems\Atom\Feature\Common\Assets\LightingPresets
- HDR file type is not yet supported, convert to EXR (see "default_iblskyboxcm.exr")
- The image processing filemask/suffix naming convention that turns a latlong input into a set of cubemaps is _iblskyboxcm
- this generates:
- - 'ibl' means radiance/irradiance lighting cubemaps (two cubemap outputs for diffuse and specular reflections, these have mipmaps processed for HDR lighting)
- - 'skybox' means generate a high-res skybox cubemaps (high resolution but no mip maps)
- if you would like to add the lighitng to the material editor, you can provide a file like this: "C:\Depot\o3de-dev\Gems\Atom\Feature\Common\Assets\LightingPresets\default.lightingpreset.azasset"
- ^ those can also be created using the 'Viewport settings panel' in the material editor

A large number of compatible lighting environments can be found here: https://polyhaven.com/

**Reply by soltsmank** [reply]
Thanks @HogJonny-AMZN

**Reply by HogJonny-AMZN** [reply]
> Thanks @HogJonny-AMZN

NP, use this page if you need more help in this area

**Comment by soltsmank** [comment]
Decide to make a small guide, based on @HogJonny-AMZN answer after successful import.

The important key here is **the file name suffix** .

There is a readme.txt located in _O3DE\21.11\Gems\Atom\Feature\Common\Assets\LightingPresets_ folder which explains how a filename affects on Asset Processor Engine (APE) to generate different output. (Explained in details every suffix possible)

Example steps for others to help:
1. Get an image that looks like flattened 3D-cube into 2D-image. The file can be opened by PTGui (www.ptgui.com) (Example: sunnyday-8k.exr)
2. Rename the file to one of possible suffixes. Lets say: '_iblskyboxcm' (Suffixes and description can be found in readme.txt, see link above) (Example: sunnyday-8k_iblskyboxcm.exr)
3. Copy renamed image to the project's folder. (APE would pick it up and output files) You can find output in Cache folder: O3DE\Projects\<YourProjectName>\Cache\
4. Use generated files (not source image) to populate input boxes for Diffuse, Specular and Skybox.
Enjoy!
